src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -492,7 +484,7 @@ public int GetHashCode(Symbol member)
                 }
 
                 if (_considerReturnType && member.GetMemberArity() == 0 && 
-                    (_typeSymbolEqualityOptions & TypeSymbolEqualityOptions.IgnoreCustomModifiers) != 0) // If it is generic, then type argument might be in return type.
+                    (_typeComparison & TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds) != 0) // If it is generic, then type argument might be in return type.
```

Comment:
Changed to `(_typeComparison & TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds) == 0`.